# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1243015322 14400
#      Fri May 22 14:02:02 2009 -0400
# Node ID 6e9ba936e98331afd8bb23f8cf3da417e48d7f91
# Parent  16f53d29049fd06078b44482bdf6df82f8930d43
interp3: don't require interpolation grid and data to have same size

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-05-22  John W. Eaton  <jwe@octave.org>
+
+	* general/interp3.m: Don't require interpolation grid to have same
+	size as data.  From Kris Thielemans <kris.thielemans@imperial.ac.uk>.
+
 2009-05-19 Carlo de Falco  <kingcrimson@tiscali.it>
 
 	* pkg/pkg.m: Fix a bug when quering only one non installed package
 	with "pkg describe".
 
 2009-05-14  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/__fdjac__.m: Support central differences.
diff --git a/scripts/general/interp3.m b/scripts/general/interp3.m
--- a/scripts/general/interp3.m
+++ b/scripts/general/interp3.m
@@ -86,17 +86,17 @@ function vi = interp3 (varargin)
 		    && nargs == (ndims (varargin{1}) + 1)))
     v = varargin{1};
     if (ndims (v) != 3)
       error ("expect 3-dimensional array of values");
     endif
     x = varargin (2:4);
     if (any (! cellfun (@isvector, x)))
       for i = 2 : 3
-	if (! size_equal (x{1}, x{i}) || ! size_equal (x{i}, v))
+	if (! size_equal (x{1}, x{i}))
 	  error ("dimensional mismatch");
 	endif
 	x{i} = permute (x{i}, [2, 1, 3]);
       endfor
       x{1} = permute (x{1}, [2, 1, 3]);
     endif
     v = permute (v, [2, 1, 3]);
     vi = ipermute (interpn (v, x{:}, method, extrapval), [2, 1, 3]);
